### DISCUSSION TITLE
shareReplay with local reference

### DISCUSSION THREAD

**Discussion by steven-rothwell** [ReactiveX/rxjs#6404] [Ideas / Feature request]
It would be nice to be able to pipe in a shareReplay(this.myReplaySubject), where this.myReplaySubject is a property of the class.  Let's say you have an http GET call that gets items and you utilize shareReplay.  You also have an http POST that posts one item.  After the successful post, it would be nice to call this.myReplaySubject.next([...this.items, item]).  It would then alert all who are subscribed to the GET of the new array of items.

I know there is a way of doing this: https://stackoverflow.com/a/67508863/787958, but there is a major flaw in this and that is you can't pass parameters to the GET call.

It would be nice to be able to share and update the subscribers are changes in the service layer.

**Comment by cartant** [comment]
In version 7, `shareReplay` is nothing more than a thin wrapper around `share`:

https://github.com/ReactiveX/rxjs/blob/cfd2c164f466266e33aef3c07958bddfae79f77e/src/internal/operators/shareReplay.ts#L137-L142

So you can do something similar, but could return some subject - from the `connector` factory - that's outside of the operator's control.

**Reply by steven-rothwell** [reply]
I tried creating my own version of share replay where instead of newing up an instance of ReplaySubject, it took a ReplaySubject as a parameter which I set as the connector.  I then would pass my local ReplaySubject as that parameter.  It worked at first and the subscribers shared the initial value.  But, if I called localReplay.next(newValue) elsewhere in my service layer, the subscribers were not notified of the newValue.

**Reply by cartant** [reply]
FWIW, if you allow the subject to complete, nexting it will be ineffectual.
